### PR TITLE
fix gzserver launch file breaking when no ros args provided

### PR DESCRIPTION
**Pull request by ihasdapie**
Fix error introduced in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1391

### CODE REVIEW

File: gazebo_ros/launch/gzserver.launch.py [hunk 1]
```diff
@@ -55,15 +55,14 @@ def generate_launch_description():
...
-        LaunchConfiguration("params_file"),
-        '--',
+        _arg_command('--ros-args', related_launch_config=LaunchConfiguration('params_file')), ' ',
+        _arg_command('--params-file', related_launch_config=LaunchConfiguration('params_file')),
+        ' ', LaunchConfiguration('params_file'), ' ',
+        _arg_command('--', related_launch_config=LaunchConfiguration('params_file')), ' ',
         LaunchConfiguration('extra_gazebo_args'),
     ]
 
+    print('--------CMD------')
```
Comment (ahcorde) [inline gazebo_ros/launch/gzserver.launch.py]: remove

Comment (ihasdapie) [reply gazebo_ros/launch/gzserver.launch.py]: Oops my bad, removed!

File: gazebo_ros/launch/gzserver.launch.py [hunk 2]
```diff
@@ -222,8 +220,11 @@ def _boolean_command(arg):
 
 
 # Add string commands if not empty
-def _arg_command(arg):
-    cmd = ['"--', arg, '" if "" != "', LaunchConfiguration(arg), '" else ""']
+def _arg_command(arg, related_launch_config=None):
```
Comment (jacobperron) [inline gazebo_ros/launch/gzserver.launch.py]: nitpick: I think a variable named `condition` might make more sense.

**Review by jacobperron — COMMENTED** [review]
Looks okay to me. It would be nice to add a launch test to this PR to validate things are working and prevent future regressions.

**Comment by ihasdapie** [comment]
@jacobperron I've pushed some basic tests to just make sure the launch works and spawns `gzserver` and `gzclient`

File: gazebo_ros/launch/gzserver.launch.py (hunk 2 above)
Comment (ihasdapie) [reply gazebo_ros/launch/gzserver.launch.py]: agreed

File: gazebo_ros/test/test_launch.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    children = current_process.children(recursive=True)
+    assert 'gzclient' in [i.name() for i in children], 'Error; gzclient not launched'
+
+
+@pytest.mark.launch(fixture=launch_gazebo)
+def test_launch_gazebo_ros(gzclient_proc, launch_context):
+    current_process = psutil.Process()
+    children = current_process.children(recursive=True)
+    names = [i.name() for i in children]
+    print(names)
```
Comment (jacobperron) [inline gazebo_ros/test/test_launch.py]:
```suggestion
```

Comment (ihasdapie) [reply gazebo_ros/test/test_launch.py]: Oops

**Review by jacobperron — COMMENTED** [review]
Though this PR fixes the regression, it doesn't solve the original issue of passing the parameters file. When I try locally, it seems the parameters file is being ignored by the gzserver process.

File: gazebo_ros/launch/gzserver.launch.py [hunk 4]
```diff
@@ -55,12 +55,10 @@ def generate_launch_description():
...
-        '--ros-args',
-        PythonExpression([
-            '"--params-file', '" if "" != "',
-            LaunchConfiguration("params_file"), '" else ""']),
-        LaunchConfiguration("params_file"),
-        '--',
+        _arg_command('--ros-args', condition=LaunchConfiguration('params_file')), ' ',
+        _arg_command('--params-file', condition=LaunchConfiguration('params_file')),
+        ' ', LaunchConfiguration('params_file'), ' ',
+        _arg_command('--', condition=LaunchConfiguration('params_file')), ' ',
```
Comment (jacobperron) [inline gazebo_ros/launch/gzserver.launch.py]: `_arg_command` is already prepending `--`. Also, the extra spaces seem to interfere with ROS argument parsing. I was able to get this to work with the following diff:

```suggestion
        _arg_command('ros-args', condition=LaunchConfiguration('params_file')),
        _arg_command('params-file', condition=LaunchConfiguration('params_file')),
        LaunchConfiguration('params_file'),
        _arg_command('', condition=LaunchConfiguration('params_file')), ' ',
```

Comment (jacobperron) [inline gazebo_ros/launch/gzserver.launch.py:61]: `_arg_command` is already prepending `--`. Also, the extra spaces seem to interfere with ROS argument parsing. I was able to get this to work with the following diff:

```suggestion
        _arg_command('ros-args', condition=LaunchConfiguration('params_file')),
        _arg_command('params-file', condition=LaunchConfiguration('params_file')),
        LaunchConfiguration('params_file'),
        _arg_command('', condition=LaunchConfiguration('params_file')), ' ',
```

Comment (jacobperron) [reply gazebo_ros/launch/gzserver.launch.py]: Using the `_arg_command` wrapper is a bit cryptic IMO, versus just using `PythonExpression` directly here. But I'm fine approving this either way.

Comment (jacobperron) [reply gazebo_ros/launch/gzserver.launch.py:61]: Using the `_arg_command` wrapper is a bit cryptic IMO, versus just using `PythonExpression` directly here. But I'm fine approving this either way.

Comment (ihasdapie) [reply gazebo_ros/launch/gzserver.launch.py]: I prefer the wrapper since it is more consistent with the rest of the file

**Review by jacobperron — APPROVED** [review]
LGTM, thanks for adding tests!

**Pull request merged**